src/zimdump.cpp

```diff
@@ -111,195 +113,117 @@ std::string urlEncode(const std::string& value, bool encodeReserved)
-      : m_file(fname),
+      : m_archive(fname),
         verbose(false)
       { }
 
     void setVerbose(bool sw = true)  { verbose = sw; }
 
     void printInfo();
-    void printNsInfo(char ch);
-    int dumpArticle(const zim::Article &a);
-    int listArticles(bool info, bool extra);
-    int listArticle(const zim::Article& article, bool extra);
-    void listArticleT(const zim::Article& article, bool extra);
-
-    zim::Article getArticleByUrl(const std::string &url);
-    zim::Article getArticle(zim::size_type idx);
+    int dumpEntry(const zim::Entry& entry);
+    int listEntries(bool info);
+    int listEntry(const zim::Entry& entry);
+    void listEntryT(const zim::Entry& entr);
 
-    bool hasNamespace(char ns);
+    zim::Entry getEntryByPath(const std::string &path);
+    zim::Entry getEntry(zim::size_type idx);
 
     void dumpFiles(const std::string& directory, bool symlinkdump, std::function<bool (const char c)> nsfilter);
 };
 
-zim::Article ZimDumper::getArticleByUrl(const std::string &url)
-{
-    return m_file.getArticleByUrl(url);
-}
-
-zim::Article ZimDumper::getArticle(zim::size_type idx)
+zim::Entry ZimDumper::getEntryByPath(const std::string& path)
 {
-    return m_file.getArticle(idx);
+    return m_archive.getEntryByPath(path);
 }
 
-bool ZimDumper::hasNamespace(char ns)
+zim::Entry ZimDumper::getEntry(zim::size_type idx)
 {
-    return m_file.hasNamespace(ns);
+    return m_archive.getEntryByPath(idx);
 }
 
 void ZimDumper::printInfo()
 {
-  std::cout << "count-articles: " << m_file.getCountArticles() << "\n";
-  if (verbose) {
-    std::string ns = m_file.getNamespaces();
-    std::cout << "namespaces: " << ns << '\n';
-    for (std::string::const_iterator it = ns.begin(); it != ns.end(); ++it) {
-      std::cout << "namespace " << *it << " size: " << m_file.getNamespaceCount(*it) << '\n';
-    }
-  }
-  std::cout << "uuid: " << m_file.getFileheader().getUuid() << "\n"
-               "article count: " << m_file.getFileheader().getArticleCount() << "\n"
-               "mime list pos: " << m_file.getFileheader().getMimeListPos() << "\n"
-               "url ptr pos: " << m_file.getFileheader().getUrlPtrPos() << "\n"
-               "title idx pos: " << m_file.getFileheader().getTitleIdxPos() << "\n"
-               "cluster count: " << m_file.getFileheader().getClusterCount() << "\n"
-               "cluster ptr pos: " << m_file.getFileheader().getClusterPtrPos() << "\n";
-  if (m_file.getFileheader().hasChecksum()) {
-    std::cout <<
-               "checksum pos: " << m_file.getFileheader().getChecksumPos() << "\n"
-               "checksum: " << m_file.getChecksum() << "\n";
+  std::cout << "count-entries: " << m_archive.getEntryCount() << "\n";
+  std::cout << "uuid: " << m_archive.getUuid() << "\n"
+            <<  "cluster count: " << m_archive.getClusterCount() << "\n";
+  if (m_archive.hasChecksum()) {
+    std::cout << "checksum: " << m_archive.getChecksum() << "\n";
   } else {
```

Comment:
For me yes, obviously :)

The different offsets of the internal arrays are pretty useless out of libzim. There is absolutely no interest to have them at `zimdump` level.
If someone want to explore a zim file, it will need to parse the binary, it can parse the header.

For the namespace, one main goal of this new api is to remove them, so it make sens also.